src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java

```diff
@@ -0,0 +1,185 @@
+                .withIdGenerator(new IdGenerator() {
+                    @Override
+                    public String generate() {
+                        return UUID.randomUUID().toString();
+                    }
+                }).store();
+        sessionStore = (SQLStore<OAuth2AuthzSession>) DataManager.getStore(AuthEndPoints.STORE);
+        sessionStore.openSync();
+        return sessionStore;
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+        OAuth2AuthzSession storedAccount = sessionStore.read(AuthEndPoints.NAME);
+        return storedAccount != null;
+    }
+
+    public void requestAccess(final Activity activity, final Callback<String> callback,
+                              final String username, final String password, final URL url) {
+
+        Exception exception;
+        String json;
+        AsyncTask<Void, Void, HeaderAndBody> task = new AsyncTask<Void, Void, HeaderAndBody>() {
+            private Exception exception;
+            String json;
+
+            @Override
+            protected HeaderAndBody doInBackground(Void... params) {
+                HeaderAndBody result;
+                try {
+                    HttpProvider provider =
+                            new HttpRestProvider(new URL(url.toString()
+                                    + AuthEndPoints.Endpoints.ACCESS));
+                    provider.setDefaultHeader("Accept", "application/json");
+                    provider.setDefaultHeader("Authorization", "Basic "
+                            + buildLoginData(username, password));
+                    result = provider.post("");
+                    Timber.e("Called with POST to: ", new String(provider.getUrl().toString()));
+                    Timber.e("Result of the POST: ", new String(result.getBody()));
+                    json = new String(result.getBody());
+                } catch (Exception e) {
+                    Timber.e(SecretStoreAuthzModule.class.getSimpleName(),
+                            "Error with Login", e);
+                    exception = e;
+                    return null;
+                }
+                return result;
+            }
+
+            @Override
+            protected void onPostExecute(HeaderAndBody headerAndBody) {
+                if (exception == null) {
+                    callback.onSuccess(new String(headerAndBody.getBody()));
+                } else {
+                    callback.onFailure(exception);
+                }
+            }
+        }.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+
+    @Override
+    public void requestAccess(final Activity activity, final Callback<String> callback) {
+        OAuth2AuthzSession storedAccount = sessionStore.read(AuthEndPoints.NAME);
+        final String token = storedAccount.getAccessToken();
+        try {
+            callback.onSuccess(token);
+        } catch (Exception e) {
```

Comment:
It's a good practice to catch only exceptions you care about, ie, those that you are able to handle. Catching the generic "Exception" means you are handling everything. In general, only code that interfaces with the end user needs to have such generic catches.

On this case, you could handle some scenarios: if the connection fails mid-flight, you might want to retry it. If you got a 404, you might want to give up immediately, as the outcome won't change anyway. Similarly, if you got a "Forbidden", it might mean that the user's password is wrong. For all cases you didn't predict, let it explode until some upper layer catches it and handle it appropriately (for instance, asking the user to check the password, or try again later, or ...).